src/backends/disk.go

```diff
@@ -74,8 +83,59 @@ func (db *DiskBackend) Save(data []byte) error {
 	return nil
 }
 
+func (f fileInfoSlice) Len() int {
+	return len(f)
+}
+
+func (f fileInfoSlice) Less(i, j int) bool {
+	return f[i].ModTime().Before(f[j].ModTime())
+}
+
+func (f fileInfoSlice) Swap(i, j int) {
+	f[i], f[j] = f[j], f[i]
+}
+
+func (db *DiskBackend) cleanOldBackups(max int) error {
+	files, err := ioutil.ReadDir(db.backupConfig.DirPath)
+	if err != nil {
+		return err
+	}
+
+	filesSorted := make(fileInfoSlice, 0, len(files))
+	for _, f := range files {
+		filesSorted = append(filesSorted, f)
+	}
+	sort.Sort(filesSorted)
+	max = min(max, len(filesSorted))
+
+	for _, f := range filesSorted[:len(filesSorted)-max] {
+		p := fmt.Sprintf("%s/%s", db.backupConfig.DirPath, f.Name())
+		if err := os.Remove(p); err != nil {
+			fmt.Println("Error", err.Error())
```

Comment:
Your point about just returning the last error is valid and we are printing the error, if one is encountered.

I'd feel better if the printed message provided a bit more context, like a `Error removing backup` prefix or something.